src/dmd/traits.d

```diff
@@ -1973,7 +1973,18 @@ Expression semanticTraits(TraitsExp e, Scope* sc)
             for (auto ns = p.cppnamespace; ns !is null; ns = ns.cppnamespace)
             {
                 ns.dsymbolSemantic(sc);
-                exps.insert(0, ns.exp);
+
+                if (ns.exp.isErrorExp())
+                    return ErrorExp.get();
+
+                auto se = ns.exp.toStringExp();
+                const sident = se.toStringz();
+                // extern(C++, (emptyTuple))
+                // struct D {}
+                // will produce a blank ident
+                if (!sident.length)
+                    continue;
```

Comment:
Couldn't this check be on `se` (before the memory allocation of `toStringz` ?

Reply:
Seems to.

Reply:
Changed.